src/backend/cpu/memory.cpp

```diff
@@ -56,29 +54,43 @@ void printMemInfo(const char *msg, const int device) {
 template<typename T>
 unique_ptr<T[], function<void(T *)>> memAlloc(const size_t &elements) {
     T *ptr = nullptr;
+    af_memory_event_pair pair = memoryManager().alloc(elements * sizeof(T), false);
+
+    af_event event;
+    af_memory_event_pair_get_event(pair, &event);
+    Event* e = getEvent(event).event;
+    if (e) e->enqueueWait(getQueue());
 
-    common::MemoryEventPair me = memoryManager().alloc(elements * sizeof(T), false);
-    if(me.e) me.e.enqueueWait(getQueue());
-    ptr = (T *)me.ptr;
+    void *inPtr;
+    af_memory_event_pair_get_ptr(pair, &inPtr);
+    ptr = (T *)inPtr;
```

Comment:
This should also get further simplified if we use std::pair as explained in one of my earlier comments. Basically where this pair object is being used, the code would be much simple without any function call.